### PR TITLE
Merge branch

### PR DESCRIPTION
**Pull request by avrhamNeeman**

### CODE REVIEW

File: src/domain/stationsList/index.js [hunk 1]
```diff
@@ -21,277 +20,192 @@
...
     const [botUrl, SetBotUrl] = useState('');
+    const [botImage, setBotImage] = useState('');
+
+    useEffect(() => {
+        dispatch({ type: 'SET_ROUTE', payload: 'stations' });
+        getAllStations();
+    }, []);
+
+    useEffect(() => {
+        if (searchInput.length >= 2) setFilteredList(stationsList.filter((station) => station.station.name.startsWith(searchInput)));
```
Comment (avrhamNeeman) [inline src/domain/stationsList/index.js:62]: Changes startWith method to Includes

File: src/domain/stationsList/index.js [hunk 2]
```diff
@@ -21,277 +20,192 @@
...
-                    factory_name: factoryDetails.name,
-                    factory_new_description: e.target.value
-                });
-                setFactoryDetails({ ...factoryDetails, description: e.target.value });
-                seteditDescription(false);
-            } catch (err) {
-                setFactoryDescription(factoryDetails.description);
+    const renderStationsOverview = () => {
+        if (stationsList?.length > 0) {
+            if (stationsList?.length === 1) {
```
Comment (avrhamNeeman) [inline src/domain/stationsList/index.js:123]: Change to if (stationsList?.length <== 2)

File: src/domain/stationsList/index.js [hunk 3]
```diff
@@ -21,277 +20,192 @@
...
     return (
-        <div className="factory-details-container">
-            {isLoading && (
-                <div className="loader-uploading">
-                    <Loader />
+        <div className="stations-details-container">
+            <div className="stations-details-header">
+                <div className="left-side">
+                    <label class="main-header-h1">
+                        Stations <label class="num-stations">{stationsList?.length > 0 ? `(${stationsList?.length})` : null}</label>
```
Comment (avrhamNeeman) [inline src/domain/stationsList/index.js:160]: change to :
{stationsList?.length > 0 && <label class="num-stations">{`(${stationsList?.length})`}</label>}

File: src/domain/stationsList/stationBoxOverview/index.js [hunk 4]
```diff
@@ -72,111 +69,101 @@ const StationBoxOverview = (props) => {
...
+            <div className="right-section">
+                <div className="station-meta">
+                    <img src={retentionImg} alt="retention" />
+                    <label className="data-labels">Retention</label>
+                    <p className="data-info">
+                        <OverflowTip
+                            text={
+                                props.station.station.retention_type === 'message_age_sec'
+                                    ? convertSecondsToDate(props.station.station.retention_value)
+                                    : props.station.station.retention_value
```
Comment (avrhamNeeman) [inline src/domain/stationsList/stationBoxOverview/index.js:95]: Move this logic to the useEffect with the switch case

File: src/domain/messageJourney/index.js [hunk 5]
```diff
@@ -39,8 +39,8 @@ import pathDomains from '../../router';
 const MessageJourney = () => {
     const [state, dispatch] = useContext(Context);
     const url = window.location.href;
-    const messageId = url.split('factories/')[1].split('/')[2];
-    const stationName = url.split('factories/')[1].split('/')[1];
+    const messageId = url.split('stations/')[1].split('/')[2];
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply src/domain/messageJourney/index.js:42]: done

File: src/domain/messageJourney/index.js [hunk 6]
```diff
@@ -39,8 +39,8 @@ import pathDomains from '../../router';
...
+    const stationName = url.split('stations/')[1].split('/')[1];
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply src/domain/messageJourney/index.js:43]: done

File: src/domain/messageJourney/index.js [hunk 7]
```diff
@@ -80,7 +80,7 @@ const MessageJourney = () => {
     }, [state.socket]);
 
     const returnBack = () => {
-        history.push(`${pathDomains.factoriesList}/${url.split('factories/')[1].split('/')[0]}/${stationName}`);
+        history.push(pathDomains.stations);
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply src/domain/messageJourney/index.js:83]: done

File: src/domain/stationsList/index.js [hunk 8]
```diff
@@ -21,277 +20,192 @@
...
 
     return (
-        <div className="factory-details-container">
-            {isLoading && (
-                <div className="loader-uploading">
-                    <Loader />
+        <div className="stations-details-container">
+            <div className="stations-details-header">
+                <div className="left-side">
+                    <label class="main-header-h1">
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply src/domain/stationsList/index.js:159]: done

File: src/components/createStationDetails/index.js [hunk 9]
```diff
@@ -170,24 +142,26 @@ const CreateStationDetails = (props) => {
...
+            }
         }
     };
 
     const createStation = async (bodyRequest) => {
         try {
             const data = await httpRequest('POST', ApiEndpoints.CREATE_STATION, bodyRequest);
             if (data) {
-                history.push(`${pathDomains.factoriesList}/${bodyRequest.factory_name}/${data.name}`);
+                history.push(pathDomains.stations);
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply src/components/createStationDetails/index.js]: done

File: src/components/createStationDetails/index.js [hunk 10]
```diff
@@ -170,24 +142,26 @@ const CreateStationDetails = (props) => {
...
             }
-        } catch (error) { }
+            props.finish();
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply src/components/createStationDetails/index.js:163]: done

**Pull request closed without merging**